### PR TITLE
📦 vroom vroom uber delivery

### PR DESCRIPTION
**Pull request by SomeOne768**
Adding the different libraries and basis classes definitions.

**TO DISCUSS:**

There are 2 kinds of libraries:
 - dynamic which can be use out of the project (but has a main, even if it is not necesseraly a problem)
 - static which is a 'framwork' and only for this project
 
 Which one do you prefer ?

### CODE REVIEW

File: Saint-Nec-Engine/CommonLib/Component.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+
+public:
+	Component() = default;
+	~Component() = default;
+	virtual void update() = 0;
+	virtual void draw() const = 0;
+};
+
+// Si pas de r�el besoin, ajouter les classes filles ici ?
```
Comment (SomeOne768) [inline Saint-Nec-Engine/CommonLib/Component.h:12]: @antonindevidal Si les classes filles ne sont pas 'énorme' et que tu es OK, on laisse tout dans le même fichier ?

File: Saint-Nec-Engine/CommonLib/GameObject.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include <vector>
+#include "Component.h"
+class GameObject
+{
+	std::vector<int> position;
```
Comment (SomeOne768) [inline Saint-Nec-Engine/CommonLib/GameObject.h:6]: J'ai utilisé des vectors, tu préfères des tableaux pour forcer 3 exactement ?

File: Saint-Nec-Engine/CommonLib/SceneManager.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#include <string>
+
+class SceneManager
+{
+	std::vector<Scene*> scenes;
+public:
+	SceneManager() = default;
+	~SceneManager() = default;
+	void addScene(Scene& scene);
+	void changeScene(std::string &name);
```
Comment (SomeOne768) [inline Saint-Nec-Engine/CommonLib/SceneManager.h:12]: Pas plus 'propre' de plutôt identifier par un ID (int) ?

File: Saint-Nec-Engine/CommonLib/Component.h (hunk 1 above)
Comment (antonindevidal) [reply Saint-Nec-Engine/CommonLib/Component.h:12]: They will be big, I just didn't think of what will be in there, so they need to be on different files

File: Saint-Nec-Engine/CommonLib/SceneManager.h (hunk 3 above)
Comment (antonindevidal) [reply Saint-Nec-Engine/CommonLib/SceneManager.h:12]: We can use an ID here if you prefer. In our case there will only a few scenes but to be futurproof, it is better to have an UUID

File: Saint-Nec-Engine/CommonLib/GameObject.h (hunk 2 above)
Comment (antonindevidal) [reply Saint-Nec-Engine/CommonLib/GameObject.h:6]: Maybe create a union struct or class to define vec3 from an array so we can call individual components (x, y, z)
There is already a vec 3 in OpenGL lib but I don't think it is a good idea to add opengl to non opengl code

File: README.md [hunk 4]
```diff
@@ -29,3 +29,16 @@ This tasks will use various implementation techniques such as particles, complex
 
 ## Sprints reviews
 
+
+### Emoji for commit
```
Comment (antonindevidal) [inline README.md:33]: Don't add this to the main readme, maybe create a new readme in doc or just keep the link on the notion

File: Saint-Nec-Engine/CommonLib/Component.cpp [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#include "pch.h"
```
Comment (antonindevidal) [inline Saint-Nec-Engine/CommonLib/Component.cpp:1]: Why is it needed?

**Review by antonindevidal — CHANGES_REQUESTED** [review]

File: Saint-Nec-Engine/CommonLib/Component.cpp (hunk 5 above)
Comment (SomeOne768) [reply Saint-Nec-Engine/CommonLib/Component.cpp:1]: Good question !

File: README.md (hunk 4 above)
Comment (SomeOne768) [reply README.md:33]: Good i'll migrate it into notion

**Pull request closed without merging**